src/platform/qt/SensorView.cpp

```diff
@@ -65,30 +65,34 @@ SensorView::SensorView(InputController* input, QWidget* parent)
 	});
 	m_input->stealFocus(this);
 	connect(m_input, &InputController::luminanceValueChanged, this, &SensorView::luminanceValueChanged);
+
+	setController(controller);
 }
 
 void SensorView::setController(std::shared_ptr<CoreController> controller) {
 	m_controller = controller;
-	connect(m_ui.timeNoOverride, &QAbstractButton::clicked, controller.get(), &CoreController::setRealTime);
-	connect(m_ui.timeFixed, &QRadioButton::clicked, [controller, this] () {
-		controller->setFixedTime(m_ui.time->dateTime().toUTC());
-	});
-	connect(m_ui.timeFakeEpoch, &QRadioButton::clicked, [controller, this] () {
-		controller->setFakeEpoch(m_ui.time->dateTime().toUTC());
-	});
-	connect(m_ui.timeOffset, &QRadioButton::clicked, [controller, this] () {
-		controller->setTimeOffset(m_ui.offsetSeconds->value());
-	});
-	connect(m_ui.offsetSeconds, qOverload<int>(&QSpinBox::valueChanged), [controller, this] (int value) {
-		if (m_ui.timeOffset->isChecked()) {
-			controller->setTimeOffset(value);
-		}
-	});
-	m_ui.timeButtons->checkedButton()->clicked();
+	if (controller) {
```

Comment:
could probably just do `if (!controller) { return; }` to reduce the diff and also reduce the indent level for the remainder of the function